src/lib/core/Optional.h

```diff
@@ -37,77 +37,119 @@ struct NullOptionalType
 };
 constexpr NullOptionalType NullOptional{};
 
+namespace detail {
+
+/// Base class that handles destruction for Optional<T>
+template <class T, typename Enable = void>
+struct OptionalBase
+{
+    bool mHasValue;
+    union
+    {
+        char mEmpty;
+        T mValue;
+    };
+
+    constexpr OptionalBase() : mHasValue(false), mEmpty(0) {}
```

Comment:
Actually, zero-initialization of `mEmpty` might be causing the code size increase. How about using `std::aligned_storage_t<sizeof(T), alignof(T)> mValue` instead? Also, I wonder if you actually need both versions as if a type is trivially desctructible then `~T()` is probably optimized out, right?

Reply:
I think using `std::monostate` (aka an empty struct) instead of `char` here should help since it will make the compiler happy that one of the variants is being initialized, without actually initializing any memory at runtime.

In my experiments I'm not seeing these destructors optimized away, but it may be something in my setup. E.g. I'm seeing global initializers emitted to register the destructor for:
```
static Optional<System::Clock::Timeout> activeThresholdTimeOverride  = NullOptional;
```

Reply:
Leaving chip::Optional as is for now